tests/test_logging/test_logging.py

```diff
@@ -5,37 +5,47 @@
 import re
 
 
-def capture(command, instream=None):
+def capture(command, instream=None, use_shell=False):
     proc = subprocess.Popen(
-        command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=instream
+        command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        stdin=instream,
+        shell=use_shell,
     )
     out, err = proc.communicate()
     return out, err, proc.returncode
 
 
-@pytest.mark.usefixtures("clean_tests_cache")
-def test_logging_to_file(python_version):
+def test_logging_to_file(tmp_path, python_version):
     """Test logging Terminal output to a log file.
     As some data will differ with each log (the timestamps, file paths, line nums etc)
     a regex substitution has been employed to replace the strings that may change with
     whitespace.
     """
-    path_basic_scene = os.path.join("tests", "tests_data", "basic_scenes.py")
-    path_output = os.path.join("tests", "tests_cache", "media_temp")
-    command = [
-        python_version,
-        "-m",
-        "manim",
-        path_basic_scene,
-        "SquareToCircle",
-        "-l",
-        "--log_to_file",
-        "--log_dir",
-        os.path.join(path_output, "logs"),
-        "--media_dir",
-        path_output,
-    ]
-    out, err, exitcode = capture(command)
+    path_basic_scene = os.path.join("tests", "test_logging", "basic_scenes.py")
+    path_output = os.path.join(tmp_path, "media_temp")
+    os.makedirs(tmp_path, exist_ok=True)
+    command = " ".join(
```

Comment:
Question: why is this command being `join`ed while the command in `tests/test_scene_rendering/test_cli_flags.py::L17` isn't?

Reply:
This is because this test sends the command to `subprocess.popen` to execute using `shell mode`, which requires that the command be sent as a string.

The reason it has to be sent as a string is due to some quirks of MacOS, which I did not delve too deep into. In short, if we don't send it over as a string, this test's temporary directories are not created, causing the test to fail. I _think_ this is because the permissions are set differently in MacOS, preventing pytest (via manim) from writing to the temporary media directories.

We don't send it over as a string in `tests/test_scene_rendering/test_cli_flags.py::L17`, because the same issue was not faced there, meaning the temporary directories were created and recognised by MacOS.

Reply:
:woozy_face: